[cleanup] Remove an empty loop

LeiW000 opened the pull request:
The loop is empty. So remove it.




<!--

If you know who should review your pull request, please assign it to that
person, otherwise the pull request would get assigned randomly.

If your pull request is for a specific language, please add the appropriate
lang label.

-->

drfloob commented:
Thanks for offering to do a cleanup. Please sign the CLA, and we can take a look.

krishnacx commented on src/core/ext/transport/chttp2/transport/chttp2_transport.cc:
```diff
@@ -2379,18 +2379,16 @@ static void read_action_locked(void* tp, grpc_error_handle error) {
     grpc_error_handle errors[3] = {error, absl::OkStatus(), absl::OkStatus()};
-    for (; i < t->read_buffer.count && errors[1] == absl::OkStatus(); i++) {
+
+    for (size_t i = 0; i < t->read_buffer.count && errors[1] == absl::OkStatus(); i++) {
```
do..while loop can be better here. Since the first iteration condition would be true anyways.

LeiW000 replied:
Good idea.

drfloob replied:
It's possible to have an empty buffer. See test errors. Please revert this bit, and just remove the unused loop in this PR.

LeiW000 commented:
@drfloob , I'm still working on signing the CLA. I have a question. Is the CLA I need to sign specific to Grpc project?

thisisnotapril commented:
@LeiW000 you appear to have signed the CLA already so all good there! https://github.com/grpc/grpc/pull/31813#issuecomment-1338560165

drfloob commented:
@LeiW000 it looks like this is failing sanity checks, you'll need to run clang-format to get the checks to pass. Let me know if you need help with that.

LeiW000 commented:
@drfloob , @thisisnotapril , thanks for your help. I have run the clang-format script and updated the PR.

@krishnacx, I have addressed your comment with the latest PR.

LeiW000 commented:
I saw some failures on the sanity tests. But I don't think it has anything to do with the change. If I'm wrong, please let me know.

drfloob commented on src/core/ext/transport/chttp2/transport/chttp2_transport.cc:
```diff
@@ -2381,16 +2381,17 @@ static void read_action_locked(void* tp, grpc_error_handle error) {
     grpc_error_handle errors[3] = {error, absl::OkStatus(), absl::OkStatus()};
-    for (; i < t->read_buffer.count && errors[1] == absl::OkStatus(); i++) {
+
+    do {
```
Commenting again for visibility: I think the test failures are due to this change, `read_buffer.count` may be 0. I recommend reverting this piece, and we can land the cleanup of the empty loop.

drfloob replied:
Looks good, thanks for the contribution. Merged.
